arviz/rcparams.py

```diff
@@ -93,11 +102,60 @@ def _validate_boolean(value):
     return value is True or value == "true"
 
 
+def make_iterable_validator(scalar_validator, length=None):
```

Comment:
I was thinking of ways to make this a little shorter and not have repeated code, then I thought about adding the `allow_none` arg and missed myself in the process. How does this look? I think it will do the same:

```
def make_iterable_validator(scalar_validator, length=None, allow_none=False, allow_auto=False):
    """Validate value is an iterable datatype."""
    # based on matplotlib's _listify_validator function

    def validate_iterable(value):
        if allow_none and (value is None or isinstance(value, str) and value.lower() == "none"):
            return None
        if isinstance(value, str):
            if allow_auto and value.lower() == "auto":
                return "auto"
            val = tuple(v.strip("([ ])") for v in value.split(",") if v.strip())  
        if np.iterable(value) and not isinstance(value, (set, frozenset)):  # note the elif changed to if
            val = tuple(scalar_validator(v) for v in value)
            if length is not None and len(val) != length:
                raise ValueError("Iterable must be of length: {}".format(length))
            return val
        raise ValueError("Only iterable values are valid")

    return validate_iterable
```